framework/include/utils/RankTwoTensor.h

```diff
@@ -279,8 +280,12 @@ class RankTwoTensorTempl : public TensorValue<T>
   RankFourTensorTempl<T> mixedProductIlJk(const RankTwoTensorTempl<T> & a) const;
 
   /// return positive projection tensor of eigen-decomposition
-  RankFourTensorTempl<T> positiveProjectionEigenDecomposition(std::vector<T> & eigval,
-                                                              RankTwoTensorTempl<T> & eigvec) const;
+  template <typename T2 = T>
+  typename std::enable_if<MooseUtils::IsLikeReal<T2>::value, RankFourTensorTempl<T>>::type
```

Comment:
I **love** this SFINAE pattern implementation. I'm using it liberally now on https://github.com/libMesh/libmesh/pull/2416.

Is this ready for review?

Reply:
Yes, I think so. By the way, we have a bunch of boostcopy::enable_if_c in moose. Should we just move to std::enable_if everywhere?

Reply:
We definitely could